enhancements/installer/compact-plus-worker.md

```diff
@@ -0,0 +1,194 @@
+---
+title: Compact + Worker deployment as day0 operation
+authors:
+  - "@flaper87"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+  - TBD
+creation-date: 2022-03-08
+last-updated: 2022-03-08
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Compact + Worker deployment as day0 operation
+
+## Summary
+
+This enhancement proposes to allow for deploying a 4-nodes cluster as a day 0
+operation, where the infrastructure workload can be deployed with HA enabled.
+The 4-nodes cluster would be composed by 3 schedulable master nodes and a
+single worker, which provide users with a highly available cluster.
+
+## Motivation
+
+There are environments where only 4 nodes are available where 4-nodes clusters
+are being deployed. We could serve OpenShift cluster administrators of these
+environments better by allowing them to deploy a 4-nodes cluster as a day0
+operation and make sure their infrastructure workload is highly available.
+
+Today, it is possible to deploy a 4-nodes cluster. The resulting cluster will
+be functional but the infrastructure workload will be run as a SingleReplica
+topology, which won't allow the various services to run as HA.
+
+To achieve the desired topology described in this enhancement,OpenShift cluster
+administrators can only deploy a 4-nodes clusters by first deploying a compact
+cluster (3 schedulable masters) and then adding a host as a day 2 operation.
+Another option is for them to This provides the same results as aimed in this
+enhancement but with a poor and slower user experience.
+
+Today, because of the current limitations in the installer, it takes
+administrators significantly longer to achieve the desired topology, when only
+4 nodes are available. Furthermore, by requiring more steps, we are making the
+process more prone to user errors.
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+- Enable the installer to deploy a 4-nodes cluster composed by 3 schedulable
+  masters and one worker.
+
+### Non-Goals
+
+- Define new infrastructure topologies
+
+- Define new architectures
+
+- Provide official support for the 4-nodes cluster deployments
+
+## Proposal
+
+- Set the `InfrastructureTopology` to have the same value as the
+  `ControlPlaneTopology` when the number of workers is 1, instead of setting its
+  value to `SingleReplica` as it's done today.
+
+- Set masters as schedulable when workers are less or equal to 1. Today,
+  masters are set to schedulable only when there are no workers available.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to be able to install a 4-nodes
```

Comment:
How many customers want to install 4-node cluster? Of those, how many want workloads on the masters?

Reply:
Our whole ZTPFW is based on 4 nodes appliance

Reply:
That's just the initial customer's use case, right? It will eventually support other configurations?

Reply:
Can we add this to the use cases with specifics of why they need or want a 4-node cluster?

Reply:
> That's just the initial customer's use case, right? It will eventually support other configurations?

It's a specific partner's use-case, at least for now. More may come with the same needs and, yes, there are others that want a different configuration already.

@staebler yes, I will add it. :+1: